### ISSUE TITLE
Todo

### ISSUE THREAD

**Issue by Divide-By-0** [Divide-By-0/whatdidyoudo#1]
Add
- Ability to sign in with GitHub to also process private repos (high)
- Have each paginated request trigger a scrape through all new organizations viewed (high)
- Include hyperlinks to the PR (medium)
- Add issues and PRs (medium) -- maybe already finished?
- Parallelize paginated request? (Low)

Done
- Have it work for non-main branches on other repos in the same org where main branch was not committed (high)
- Add the streaming ChatGPT summary to a edge function with a vercel template (high)
- Add loading spinners (low)

Done
- Fix i.e. yearly request to go most recent first (high)
- Have it work for organization names too (high)
- Work for non-main branches on repos where main had at least one commit (high)

**Comment by rmccrear** [comment]
If you allow access to private repos, is there any concern with leaking private information? 

If you have commits to a private repo, and share the whatyoudid, it will show the commit comments and the AI summary.

Is this going to be OK, or is there something that would need to be done to mitigate leaking info from private repos?

**Comment by Divide-By-0** [comment]
I think that's all right, as long as we're upfront with the user about what we're accessing and why!

**Comment by rmccrear** [comment]
OK. And the user can review what it shows before posting publicly.

**Comment by Divide-By-0** [comment]
Yeah definitely, nothing is posted publicly by default, they have to click share in order for that to happen, maybe we can make it even clearer with a pop-up or something?
